identity-account/src/tests/updates.rs

```diff
@@ -0,0 +1,656 @@
+
+  Ok(())
+}
+
+#[tokio::test]
+async fn test_create_scoped_method() -> Result<()> {
+  for scope in &[
+    MethodScope::Authentication,
+    MethodScope::AssertionMethod,
+    MethodScope::KeyAgreement,
+    MethodScope::CapabilityInvocation,
+    MethodScope::CapabilityDelegation,
+  ] {
+    let mut account = Account::create_identity(account_setup(), IdentitySetup::default()).await?;
+
+    let fragment = "#key-1".to_owned();
+
+    let update: Update = Update::CreateMethod {
+      scope: *scope,
+      method_secret: None,
+      type_: MethodType::Ed25519VerificationKey2018,
+      fragment: fragment.clone(),
+    };
+
+    account.process_update(update).await?;
+
+    let state: &IdentityState = account.state();
+
+    assert_eq!(state.document().as_document().verification_relationships().count(), 2);
+
+    assert_eq!(state.document().as_document().methods().count(), 2);
+
+    let mut relative_url = RelativeDIDUrl::new();
+    relative_url.set_fragment(Some(&fragment)).unwrap();
+
+    let core_doc = state.document().as_document();
+
+    let contains = match scope {
+      MethodScope::Authentication => core_doc
+        .authentication()
+        .iter()
+        .any(|method_ref| method_ref.id().url() == &relative_url),
+      MethodScope::AssertionMethod => core_doc
+        .assertion_method()
+        .iter()
+        .any(|method_ref| method_ref.id().url() == &relative_url),
+      MethodScope::KeyAgreement => core_doc
+        .key_agreement()
+        .iter()
+        .any(|method_ref| method_ref.id().url() == &relative_url),
+      MethodScope::CapabilityDelegation => core_doc
+        .capability_delegation()
+        .iter()
+        .any(|method_ref| method_ref.id().url() == &relative_url),
+      MethodScope::CapabilityInvocation => core_doc
+        .capability_invocation()
+        .iter()
+        .any(|method_ref| method_ref.id().url() == &relative_url),
+      _ => unreachable!(),
+    };
+
+    assert!(contains);
+  }
+
+  Ok(())
+}
+
+#[tokio::test]
+async fn test_create_method_duplicate_fragment() -> Result<()> {
+  let mut account_setup = account_setup();
+  account_setup.config = account_setup.config.autopublish(false);
+
+  let mut account = Account::create_identity(account_setup, IdentitySetup::default()).await?;
+
+  let update: Update = Update::CreateMethod {
+    scope: MethodScope::default(),
+    method_secret: None,
+    type_: MethodType::Ed25519VerificationKey2018,
+    fragment: "key-1".to_owned(),
+  };
+
+  account.process_update(update.clone()).await?;
+
+  let output = account.process_update(update.clone()).await;
+
+  // Attempting to add a method with the same fragment in the same int and diff generation.
+  assert!(matches!(
+    output.unwrap_err(),
+    Error::UpdateError(UpdateError::DuplicateKeyLocation(_)),
+  ));
+
+  // Fake publishing by incrementing the generation.
+  account.state_mut_unchecked().increment_generation().unwrap();
```

Comment:
Can we instead set `testmode` to true on the `Account` and call `publish_updates`? Not critical if generations are being removed.